### ISSUE TITLE
in an application where characters are automatically entered, nvda informs the wrongly typed character, and does not announce the insertion of automatically entered characters 

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#12661]
### Steps to reproduce:
unfortunately I cannot inform a step by step to replicate this problem, it occurs in an internal development application of the company where I work, without external access.
I report below in the hope that someone can replicate it in another application.
this is a java application, I don't know if this problem only occurs in java applications or if it occurs in other programs in other languages.
In this application there is a field to enter the date, in the format:
2 digits for day, 2 digits for month and 4 digits for year.
if I type 2 digits for the day and press backspace to delete the last digit of the day typed nvda will announce /, actually 2 characters were deleted the slash that the system automatically starts, but the second character of the day was also deleted, which the nvda does not advertise.
Another problem, when I type 2 characters for the day the app inserts / automatically, but nvda doesn't announce the / 
### Actual behavior:
when pressing backspace nvda announces the deletion of / that was entered by the app, and when we enter the date in a running form 2 digits for day 2 days for month and 4 digits for year nvda does not announce the / automatically entered by the application 

### Expected behavior:
i expect nvda when erasing with backspace announces the / entered by the app, but also announces the actually typed character.
when the date is entered, the / automatically entered by the application must be announced 
#### NVDA installed/portable/running from source:
instaled
#### NVDA version:
nvda.exe, NVDA alpha-23364,b7dd32fc

#### Windows version:
10.0.19044.1147
#### Name and version of other software in use when reproducing the issue:
app java java.exe, Java(TM) Platform SE 8 8.0.3010.9

#### Does the issue still occur after restarting your computer?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no
#### If add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes
2 logs:
[sl.txt](https://github.com/nvaccess/nvda/files/6852005/sl.txt)
[sl1.txt](https://github.com/nvaccess/nvda/files/6852006/sl1.txt)

**Comment by Adriani90** [comment]
@fernando-jose-silva when you enter 2 digits for day, what does NVDA announce after that? Does it announce something like "blank edit field"? If yes, then it seems the day, month and year are placed in different edit fields and the slash is just a sign appearing automatically between the edit fields but not in the fields themselves.

If you navigate with arrow keys left and right, does NVDA read the slash? Can you select the slash with shift + arrow key and delete it with the delete key?

**Comment by fernando-jose-silva** [comment]
[sl3.txt](https://github.com/nvaccess/nvda/files/6994105/sl3.txt)
when I enter the 2 characters of the day nvda doesn't announce anything.
in this case I would like it to announce that the / has been entered by the app
there is only one edit field, and the / is inserted into this single edit field when the number of characters is entered, for example the application expects you to enter 2 characters for day, by doing this the / is automatically inserted into this single text field .
when navigating with the left and right arrows nvda does not announce the existence of /.
at this point you would expect nvda to announce itself to /.
when I go with the arrow keys to the second character of the day and press shift right arrow to select the second character of the day
nvda announces the selected character and /
which in my view seems to be correct.
Now if I select the second character of the day and press delete, something that I've never seen in other editing fields also happens, the application leaves a space between the last character of the day and the first character of the month.
I will attach a log with the spoken steps like this.

**Comment by Adriani90** [comment]
Hmm, I think the slash might be not treated like a single character in that application, that's why NVDA might not pronounce it separately. cc: @leonardder or @JulienCochuyt maybe you have some suggestions here?

**Comment by Adriani90** [comment]
@fernando-jose-silva are you still having this issue in NVDA 2024.4.1?

**Comment by fernando-jose-silva** [comment]
Yes, I can still play it.
Thank you very much.
